### PR TITLE
Add ushahidi/kohana-validation package and use that

### PR DESCRIPTION
**Pull request by rjmackay**
This replaces the dependencies on Kohana Core.

- Update Validators to refer to namespaced Valid and Validation class
- Fix DI for ValidationEngineTrait
- Set up translations
- Move error messages to laravel language files

Test checklist:
- [ ] Check travis build runs with only failures in dataproviders

Refs #1372 

Ping @ushahidi/platform

### CODE REVIEW

**Comment by rjmackay** [comment]
@willdoran can you review?
Validation package is over here: https://github.com/ushahidi/kohana-validation
I'm not super happy that I had to include the `Arr` helper but it was the fastest path to done.

File: src/App/Validator/KohanaValidationEngine.php [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    public function __construct(Translator $translator)
+    {
+        $this->translator = $translator;
+
+        parent::__construct([], $this->getTranslationCallback());
+    }
+
+    public function getTranslationCallback()
+    {
+        return function ($file, $field, $error = null) {
```
Comment (rjmackay) [inline src/App/Validator/KohanaValidationEngine.php]: This is probably the ugliest bit. We pass a callback into the KohanaValidation lib for translating messages, and do some wrangling here to connect that to lumen's translator.

File: src/Core/Tool/ValidationEngineTrait.php [hunk 2]
```diff
@@ -27,8 +26,8 @@
      * @param  $validation_factory
      * @return void
      */
-    public function setValidation(InstanceFactory $validation_factory)
+    public function setValidation(ValidationEngine $validation_engine)
```
Comment (rjmackay) [inline src/Core/Tool/ValidationEngineTrait.php]: Not sure *why* this was set up so weirdly. Cleaner now

**Comment by willdoran** [comment]
@rjmackay This is to stablise validation during transition?

**Comment by rjmackay** [comment]
> @rjmackay This is to stablise validation during transition?

Yup. We need to drop Kohana ASAP, so this just keeps the couple of classes we need in the interrim.

**Pull request merged**